sorc/machine-setup.sh

```diff
@@ -0,0 +1,62 @@
+# Create a test function for sh vs. bash detection.  The name is
+# randomly generated to reduce the chances of name collision.
+__ms_function_name="setup__test_function__$$"
+eval "$__ms_function_name() { /bin/true ; }"
+
+# Determine which shell we are using
+__ms_ksh_test=$( eval '__text="text" ; if [[ $__text =~ ^(t).* ]] ; then printf "%s" ${.sh.match[1]} ; fi' 2> /dev/null | cat )
+__ms_bash_test=$( eval 'if ( set | grep '$__ms_function_name' | grep -v name > /dev/null 2>&1 ) ; then echo t ; fi ' 2> /dev/null | cat )
+
+if [[ ! -z "$__ms_ksh_test" ]] ; then
+    __ms_shell=ksh
+elif [[ ! -z "$__ms_bash_test" ]] ; then
+    __ms_shell=bash
+else
+    # Not bash or ksh, so assume sh.
+    __ms_shell=sh
+fi
+
+target=""
+USERNAME=`echo $LOGNAME | awk '{ print tolower($0)'}`
+
+if [[ -d /lfs5 ]] ; then
+     # We are on NOAA Jet
+    if ( ! eval module help > /dev/null 2>&1 ) ; then
+        echo load the module command 1>&2
+        source /apps/lmod/lmod/init/$__ms_shell
+    fi
+    target=jet
+    module purge
+elif [[ -d /scratch1/NCEPDEV ]] ; then
+    # We are on NOAA Hera
+    if ( ! eval module help > /dev/null 2>&1 ) ; then
+        echo load the module command 1>&2
+        source /apps/lmod/lmod/init/$__ms_shell
+    fi
+    target=hera
+    module purge
+elif [[ -d /work2/noaa ]] ; then
+    # We are on MSU Orion
+    if ( ! eval module help > /dev/null 2>&1 ) ; then
+        echo load the module command 1>&2
+        source /apps/lmod/lmod/init/$__ms_shell
+    fi
+    target=orion || hercules
+    module purge
+    module use /apps/modulefiles/core
+    module use /apps/contrib/modulefiles
+    module use /apps/contrib/NCEPLIBS/lib/modulefiles
+    module use /apps/contrib/NCEPLIBS/orion/modulefiles
```

Comment:
Determining if you are on Orion or Hercules can be tricky.  Here is a method we have used that works pretty well:

```bash
elif [[ -d /work2/noaa ]]; then
  # We are on MSU Orion or Hercules
  mount=$(findmnt -n -o SOURCE /home)
  if [[ ${mount} =~ "hercules" ]]; then
    MACHINE_ID=hercules
  else
    MACHINE_ID=orion
  fi
```